opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
Eddie, more complexly, what if one user sets

ConceptNode "poop" <.0003>

and another sets

SatisfyingSet <.99>
     InheritanceLink
           $X
           ConceptNode "poop"


These are mathematically contradictory, just in a less obvious way
than the example you're citing...

I agree it's awkward that we have a redundant representation wherein the TV
on a 0-ary predicate has the same semantics as the TV on the EvalLink
wrapping the 0-ary predicate.  However, I think that for now it is better
just to be careful to use these consistently, than to undertake a major
change of the representational mechanisms used with PLN, Atomese, etc.
There is no logical inconsistency here, just a minor redundancy...

ben



On Mon, Aug 1, 2016 at 11:38 AM, Eddie Monroe <notifications@github.com>
wrote:

> In opencog/openpsi/utilities.scm
> <https://github.com/opencog/opencog/pull/2334#discussion_r73010118>:
>
> >  				   (equal? type 'DefinedPredicateNode))
> >  				(set! result (cog-evaluate! (Evaluation entity (List)))))
> > -			(if (or (equal? type 'SchemaNode)
> > -            		(equal? type 'GroundedSchemaNode)
> > -            	    (equal? type 'DefinedSchemaeNode))
> > +			(if (equal? type 'PredicateNode)
> > +				(set! result (cog-tv entity)))
>
> Well since they are both valid representations with no meaningful
> difference, what if one process, e.g., ros rig control, uses the
> PredicateNode TV to store the value, and another process, e.g., PLN, uses
> an EvaluationLink TV to store the value. Neither "user" has made an error,
> and we now have different values.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencog/opencog/pull/2334/files/255ac40a46ca72d1fe5b4f88e3959a99a2c6658d#r73010118>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFolXFxI4--4MgbO1V3V5H15Lnuq-lDTks5qbiDugaJpZM4JYxpE>
> .
>



-- 
Ben Goertzel, PhD
http://goertzel.org

Super-benevolent super-intelligence is the thought the Global Brain is
currently struggling to form...